### PR TITLE
Blogpost about testing

### PR DESCRIPTION
**Pull request by KBanecka**

### CODE REVIEW

File: blog/content/post/2021-02-05-testing.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  - App Dev
+  - Security
+  - Manufacturing
+
+---
+
+#### The Importance of test automation
+
+
+ <div style="text-align: justify"> It is common understanding that software, or – in our case – firmware must be
```
Comment (miczyg1) [inline blog/content/post/2021-02-05-testing.md:30]: Can't we instruct hugo to justify the paragraphs? @artur-rs ?

File: blog/content/post/2021-02-05-testing.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+thoroughly tested before releasing it to the public. Without doing that we risk
+leaving security vulnerabilities, user-experience destroying bugs, and less
```
Comment (miczyg1) [inline blog/content/post/2021-02-05-testing.md:32]:
```suggestion
leaving security vulnerabilities, user-experience destroying bugs and less
```

File: blog/content/post/2021-02-05-testing.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+![testing_chart](/img/test2.png)
+
+#### Why is it so important?
+<div style="text-align: justify">
+Trust and confidence are critical to the success of a firmware product. There
+are two main methods for achieving it: transparency (users can see how the
+testing model is performed) and hard validation (users can track how deeply the
+product is tested). The greatest result brings joining those two methods.
+Publishing the results of hard validation gives the purest quality proof of the
+product deliverable. This is how we are performing in 3mdeb, to could describe our
```
Comment (miczyg1) [inline blog/content/post/2021-02-05-testing.md:98]:
```suggestion
product deliverable. This is how we are performing in 3mdeb, to describe our
```
?

File: blog/content/post/2021-02-05-testing.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+products as secure, and optimized.
```
Comment (miczyg1) [inline blog/content/post/2021-02-05-testing.md:99]:
```suggestion
products as secure and optimized.
```

File: blog/content/post/2021-02-05-testing.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+#### The Importance of test automation
+
+
+ <div style="text-align: justify"> It is common understanding that software, or – in our case – firmware must be
+thoroughly tested before releasing it to the public. Without doing that we risk
+leaving security vulnerabilities, user-experience destroying bugs, and less
+optimal performance. While it may seem obvious what should be done, more
+specific glimpse proves that there is more than one possible approach to
+testing, and the choice clearly determines the further results. The
```
Comment (miczyg1) [inline blog/content/post/2021-02-05-testing.md:35]:
```suggestion
testing and the choice clearly determines the further results. The
```

File: blog/content/post/2021-02-05-testing.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+The whole process may be processed in a much more convenient way. Tests can be
+run by the developer, who needs only a brief information about the testing
+environment. Clear documentation and ready-to-use physical platforms with system
+of automatic availability management make it possible to run both, the single
+tests and complete suites, concerning a set of functionalities that must be
+checked. In result we receive immediate feedback about the changes made and the
+features that were added.
+
+An automated testing infrastructure can be paired with constant integration, and
```
Comment (miczyg1) [inline blog/content/post/2021-02-05-testing.md:72]:
```suggestion
An automated testing infrastructure can be paired with constant integration and
```

File: blog/content/post/2021-02-05-testing.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+delivery pipeline, to enhance the overall performance. We successfully use such
+approach when testing our projects. It all begins when most recent changes for
+the release are ready in our core repository and marked as a
+release. From this point, the automatic process of building usable binaries
+begins, signing it with authenticity proving keys, testing, and delivering the
```
Comment (miczyg1) [inline blog/content/post/2021-02-05-testing.md:77]:
```suggestion
begins signing it with authenticity proving keys, testing, and delivering the
```

File: blog/content/post/2021-02-05-testing.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+Publishing the results of hard validation gives the purest quality proof of the
+product deliverable. This is how we are performing in 3mdeb, to could describe our
+products as secure, and optimized.
+
+Usually the testing process is ended when the results are made available
+for public eyes. We could end this post here as well, but there is something
+more. For the analysis it is important, how the tests are
+presented. Usually the results are stored in the charts from publicly available
+spreadsheets, serving the informing role only. This process can be far improved,
+and automated, and this is what we did. While waiting for our automated tests to
```
Comment (miczyg1) [inline blog/content/post/2021-02-05-testing.md:106]:
```suggestion
and automated and this is what we did. While waiting for our automated tests to
```

File: blog/content/post/2021-02-05-testing.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+are two main methods for achieving it: transparency (users can see how the
+testing model is performed) and hard validation (users can track how deeply the
+product is tested). The greatest result brings joining those two methods.
+Publishing the results of hard validation gives the purest quality proof of the
+product deliverable. This is how we are performing in 3mdeb, to could describe our
+products as secure, and optimized.
+
+Usually the testing process is ended when the results are made available
+for public eyes. We could end this post here as well, but there is something
+more. For the analysis it is important, how the tests are
```
Comment (miczyg1) [inline blog/content/post/2021-02-05-testing.md:103]: Inocrrect wrapping

File: blog/content/post/2021-02-05-testing.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+presented. Usually the results are stored in the charts from publicly available
+spreadsheets, serving the informing role only. This process can be far improved,
+and automated, and this is what we did. While waiting for our automated tests to
+be processed, we have created Dasharo Transparent Validation. System, that
+automatically presents the results in a clean way that is easy to get, and have
```
Comment (miczyg1) [inline blog/content/post/2021-02-05-testing.md:108]:
```suggestion
automatically presents the results in a clean way that is easy to get. It also has
```

This sentence would be too long

File: blog/content/post/2021-02-05-testing.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+many available filters to customize the view for presenting the exact data
+needed, with support for charts embedabble on a website or repository readme
```
Comment (miczyg1) [inline blog/content/post/2021-02-05-testing.md:110]:
```suggestion
needed, with support for charts embeddable on a website or repository readme
```

File: blog/content/post/2021-02-05-testing.md [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+notes. You will hear more about Dasharo Transparent Validation soon.
+</div>
+
+![testing_chart](/img/test3.png)
+
+
+## Summary
+
+ We think that our team can help you improving the security of your firmware.
```
Comment (miczyg1) [inline blog/content/post/2021-02-05-testing.md:119]: This is not our standard summary. Secondly we should avoid "we know better what you think" approach. Thirdly we simply can, we have no doubts. Please revert to the old summary.

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: blog/content/post/2021-02-05-testing.md [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+layout: post
+published: true
+date: 2021-02-05
+archives: "2021"
+
+tags:
+  - testing
+  - firmware
+categories:
+  - Firmware
```
Comment (artur-rs) [inline blog/content/post/2021-02-05-testing.md:17]: @KBanecka do we plan to use all categories for this post?

File: blog/content/post/2021-02-05-testing.md (hunk 1 above)
Comment (artur-rs) [reply blog/content/post/2021-02-05-testing.md:30]: @miczyg1 we can add simple css change:
```
.post-content p {
    text-align: justify;
}
```
but do we want that in all posts? @KBanecka ?

Comment (artur-rs) [reply blog/content/post/2021-02-05-testing.md:30]: @KBanecka if you want to test that, find `bulma.css` file and add or replace above method in the code, then do the local preview of the blog post

File: blog/content/post/2021-02-05-testing.md [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+of automatic availability management make it possible to run both, the single
+tests and complete suites, concerning a set of functionalities that must be
+checked. In result we receive immediate feedback about the changes made and the
+features that were added.
+
+An automated testing infrastructure can be paired with constant integration, and
+delivery pipeline, to enhance the overall performance. We successfully use such
+approach when testing our projects. It all begins when most recent changes for
+the release are ready in our core repository and marked as a
+release. From this point, the automatic process of building usable binaries
```
Comment (artur-rs) [inline blog/content/post/2021-02-05-testing.md:76]: @KBanecka not "release candidate"? Always marked as `_rc1` etc.?

**Comment by artur-rs** [comment]
@KBanecka Why there is no word about RTE? Links to this product, graphics from our internal rte tests repo (tests-infrastructure), even link to the RTE datasheet?

File: blog/content/post/2021-02-05-testing.md (hunk 1 above)
Comment (KBanecka) [reply blog/content/post/2021-02-05-testing.md:30]: dodałam zmianę

File: blog/content/post/2021-02-05-testing.md (hunk 13 above)
Comment (KBanecka) [reply blog/content/post/2021-02-05-testing.md:19]: additrional categories removed

File: blog/content/post/2021-02-05-testing.md [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+time-consuming tasks are performing automatically, it leaves developers more
+time for truly creative work on delivering the highest quality of product.
+
+Test verification process needs understanding of the tested functionality and
+further interpretation of the results. In case of the automated testing, the
+output provides result and definition for getting the necessary information for
+users, clients and developers. When the complete regression testing results for
+the given release are ready, the next step is to make them accessible and easy
+to understand.
+
```
Comment (artur-rs) [inline blog/content/post/2021-02-05-testing.md:100]: @KBanecka please add also note about the RTE tag on our blog: https://blog.3mdeb.com/tags/rte/

File: blog/content/post/2021-02-05-testing.md [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+#### Remote Testing Environment
+
+As we are performing our test on hardware we had to found a way to automate
+management and operations on physical platforms. To meet that goal we have
+created Remote Testing Environment - a Orange Pi Zero hat that gives us
+ability to remotely control power, simulate pressing of buttons like power on
+or reset, read serial output and flash firmware using physical cable connection
+that can unbrick device without any manual intervention. This setup has a
+corresponding configuration file for every platform model which allow for
+smooth remote test execution. You can find [RTE](https://jira-3mdeb.atlassian.net/browse/SHA-195?focusedCommentId=55760&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-55760) in our shop.
```
Comment (artur-rs) [inline blog/content/post/2021-02-05-testing.md:80]: @KBanecka link to be replaced for correct one

Comment (KBanecka) [reply blog/content/post/2021-02-05-testing.md]: done

File: blog/content/post/2021-02-05-testing.md (hunk 15 above)
Comment (KBanecka) [reply blog/content/post/2021-02-05-testing.md:103]: done

**Review by artur-rs — APPROVED** [review]

**Pull request merged**